tests/unit/records/test_api_views.py

```diff
@@ -141,3 +141,30 @@ def test_custom_search(es, api, json_headers, record_with_bucket,
                         custom=query),
                 headers=json_headers)
             assert len(res.json) == result
+
+
+@pytest.mark.parametrize(('query', 'result'), [
+    ("true", "true"),
+    ("True", "true"),
+    ("1", "true"),
+    ("1234", "1234"),
+    ("test", "test"),
+])
+def test_apply_version_filters(es, api, json_headers, record_with_bucket,
+                               custom_metadata, query, result):
+    """Test record version filters to search."""
+    pid, record = record_with_bucket
+    record['custom'] = custom_metadata
+    RecordIndexer().index(record)
+    current_search.flush_and_refresh(index='records')
+    with api.test_request_context():
+        with api.test_client() as client:
+            res = client.get(
+                    url_for('invenio_records_rest.recid_list',
+                            all_versions=query),
+                    headers=json_headers)
+            assert res.headers["Link"] == '<http://localhost/' \
```

Comment:
Actually, an easier place to check is the response's JSON body (I think via `res.json`), and parse the `links.self` key's value [using `urllib`](https://stackoverflow.com/a/39841923/2099726) to compare.

Reply:
I have changed to this:
```python
parsed_url = urlparse(res.headers["Link"])
parsed_url = parse_qs(parsed_url.query)
assert parsed_url["all_versions"][0] == result
```

The `res.json` outputted `[]`